### ISSUE TITLE
VAAPI support

### ISSUE THREAD

**Issue by ArchangeGabriel** [pdfpc/pdfpc#389]
**Expected Behaviour**

Video playback should use VAAPI.

**Actual Behaviour**

Uses CPU instead.

**Steps to reproduce**

1. Launch any PDF presentation with integrated video.
2. Observe low framerate/high CPU usage.

pdfpc version: 4.2.1
Used Distribution (GTK version, vala version): ArchLinux (3.24.1+8+g220f77d8c1, 0.42.3)
Display Server (X/Wayland): X
HiDPI Setting (yes/no): yes

**Comment by AndreasBilke** [comment]
We use `gtksink` for rendering. If I understand it correctly, `gtksink` does not use hardware accelerated rendering, but it works everywhere.

There is `gtkglsink` for faster rendering, but it is not guaranteed to work in every situation.

See https://gstreamer.freedesktop.org/releases/1.14/ , section "Qt QML and GTK plugins moved from -bad to -good".

I prefer the safe solution over better performance.

If you are willing to make this configurable, feel free to submit a patch.

**Comment by ArchangeGabriel** [comment]
Well for vaapi I think one should use `vaapidecodebin` and `vaapisink`. I don’t have the necessary knowledge to provide a patch for adding this as a configurable option, but at least I would like to keep this issue open as a feature request.

**Comment by fnevgeny** [comment]
I agree with @AndreasBilke about stability vs performance. Ironing out video-related bugs should be given a higher priority.
